dtonator/src/main/java/com/bizo/dtonator/GenerateEnum.java

```diff
@@ -44,6 +52,29 @@ private void addEnumValues() {
     }
   }
 
+  private void addEnumGetters() {
+    try {
+      final Class<?> clazz = Class.forName(dto.getDomainType());
+      for (final Field field : clazz.getDeclaredFields()) {
+        if (!field.isEnumConstant() && !field.isSynthetic()) {
+          final GMethod getter = gc.getMethod("get{}", capitalize(field.getName())).returnType(field.getType());
+          getter.body.line("switch (this) {");
+          for (final String value : dto.getEnumValues()) {
+            @SuppressWarnings({ "unchecked", "rawtypes" })
+            final Enum enumClass = Enum.valueOf((Class<? extends Enum>) clazz, value);
+            final Object returnValue = PropertyUtils.getProperty(enumClass, field.getName());
+            getter.body.line("_ case {}: return {};", value, returnValue instanceof String ? "\"" + returnValue + "\"" : returnValue);
+          }
+          getter.body.line("}");
+          getter.body.line("return null;");
+        }
+      }
+    } catch (SecurityException | ClassNotFoundException | IllegalAccessException | InvocationTargetException | IllegalArgumentException
```

Comment:
I'd be tempted to just put a "catch Exception" around line 65, e.g. when we try and call the getter, if the getter fails for any reason, skip it, but then continue on to the next field/getter, and try to generate a method for it.